packages/tenant-process/src/services/tenantService.ts

```diff
@@ -721,18 +779,28 @@ export function tenantServiceBuilder(
           attr.type === tenantAttributeType.VERIFIED && attr.id === attribute.id
       );
 
+      if (delegateId && delegateId !== organizationId) {
+        throw error;
+      }
+
+      if (!delegateId && organizationId !== agreement.producerId) {
+        throw error;
+      }
+
```

Comment:
I think this should be inside `assertVerifiedAttributeOperationAllowed` since it is also a shared check between the verification and revocation action

Reply:
It might be worth looking into the reason the test fails, imho

Reply:
The reason is simple, it fails because other errors are expected before these new checks. I'll try to see if there is a way move this checks inside the method but in a way it doesn't require to modify too much in tests

Reply:
Ok, I think I'm crazy, I retried to move the check inside the method and now it works as expected. 🤷‍♂️